File: app/src/main/java/com/mytlogos/enterprise/background/room/Converters.java
package com.mytlogos.enterprise.background.room;

import androidx.room.TypeConverter;

import com.mytlogos.enterprise.Formatter;

import org.joda.time.DateTime;

public class Converters {

    @TypeConverter
    public static DateTime fromTimeStamp(String s) {
        return s == null ? null : Formatter.parseDateTime(s);
    }

    @TypeConverter
    public static String fromDateTime(DateTime dateTime) {
        return dateTime == null ? null : dateTime.toInstant().toString();
    }
}

File: app/src/main/java/com/mytlogos/enterprise/background/room/Converters.kt
package com.mytlogos.enterprise.background.room

import androidx.room.TypeConverter
import com.mytlogos.enterprise.formatDateTimeIso
import com.mytlogos.enterprise.parseDateTime
import org.joda.time.DateTime

/**
 * Converter Class for [AbstractDatabase].
 */
object Converters {
    @TypeConverter
    fun fromTimeStamp(s: String?): DateTime? {
        return s?.parseDateTime()
    }

    @TypeConverter
    fun fromDateTime(dateTime: DateTime?): String? {
        return dateTime?.formatDateTimeIso()
    }
}